src/ComServerHelpers/WinRtServer.cs

```diff
@@ -0,0 +1,291 @@
+        return true;
+    }
+
+    /// <summary>
+    /// Unregister an activation factory with the server.
+    /// </summary>
+    /// <param name="factory">The activation factory to unregister.</param>
+    /// <returns><see langword="true"/> if <paramref name="factory"/> was unregistered; otherwise, <see langword="false"/>.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="factory"/> is <see langword="null"/>.</exception>
+    /// <exception cref="InvalidOperationException">The server is running.</exception>
+    public bool UnregisterActivationFactory(BaseActivationFactory factory)
+    {
+        ObjectDisposedException.ThrowIf(IsDisposed, this);
+        if (IsRunning)
+        {
+            throw new InvalidOperationException("Can only remove activation factories when server is not running");
+        }
+        ArgumentNullException.ThrowIfNull(factory);
+
+        return factories.Remove(factory.ActivatableClassId);
+    }
+
+    private unsafe HRESULT ActivationFactoryCallback(HSTRING activatableClassId, IActivationFactory** factory)
+    {
+        if (activatableClassId == HSTRING.Null || factory is null)
+        {
+            return HRESULT.E_INVALIDARG;
+        }
+
+        if (!factories.TryGetValue(activatableClassId.AsString(), out var managedFactory))
+        {
+            factory = null;
+            return HRESULT.E_NOINTERFACE;
+        }
+
+        var unknown = comWrappers.GetOrCreateComInterfaceForObject(new BaseActivationFactoryWrapper(managedFactory.Factory, managedFactory.Wrapper), CreateComInterfaceFlags.None);
+        var hr = (HRESULT)Marshal.QueryInterface(unknown, in global::Windows.Win32.System.WinRT.IActivationFactory.IID_Guid, out nint ppv);
+        *factory = (IActivationFactory*)ppv;
+        if (unknown != 0)
+        {
+            Marshal.Release(unknown);
+        }
+
+        return hr;
+    }
+
+    /// <summary>
+    /// Gets a value indicating whether the instance is disposed.
+    /// </summary>
+    public bool IsDisposed
+    {
+        get;
+        private set;
+    }
+
+    /// <summary>
+    /// Gets a value indicating whether the server is running.
+    /// </summary>
+    public bool IsRunning => registrationCookie != 0;
+
+    /// <summary>
+    /// Starts the server.
+    /// </summary>
+    /// <remarks>Calling <see cref="Start"/> is non-blocking.</remarks>
+    public unsafe void Start()
+    {
+        ObjectDisposedException.ThrowIf(IsDisposed, this);
+        if (IsRunning)
+        {
+            return;
+        }
+
+        string[] managedActivatableClassIds = [.. factories.Keys];
+        HSTRING* activatableClassIds = null;
+        delegate* unmanaged[Stdcall]<HSTRING, IActivationFactory**, HRESULT>* activationFactoryCallbacks = null;
+        try
+        {
+            activatableClassIds = (HSTRING*)Marshal.AllocHGlobal(sizeof(HSTRING) * managedActivatableClassIds.Length);
```

Comment:
Instead of Alloc from Marshal, we typically make use of `NativeMemory.Alloc` in our .NET 8 and later projections given we seen it being faster.  Should we update our 3 uses here?

Reply:
Updated.